apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/StylingView.tsx

```diff
@@ -0,0 +1,133 @@
+import BackgroundStylingCard from "@/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/BackgroundStylingCard";
+import CardStylingSettings from "@/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/CardStylingSettings";
+import FormStylingSettings from "@/app/(app)/environments/[environmentId]/surveys/[surveyId]/edit/components/FormStylingSettings";
+import { RotateCcwIcon } from "lucide-react";
+import Link from "next/link";
+import React, { useEffect, useMemo, useState } from "react";
+import toast from "react-hot-toast";
+
+import { TEnvironment } from "@formbricks/types/environment";
+import { TProduct } from "@formbricks/types/product";
+import { TSurvey } from "@formbricks/types/surveys";
+import { Button } from "@formbricks/ui/Button";
+import { Switch } from "@formbricks/ui/Switch";
+
+type StylingViewProps = {
+  environment: TEnvironment;
+  product: TProduct;
+  localSurvey: TSurvey;
+  setLocalSurvey: React.Dispatch<React.SetStateAction<TSurvey>>;
+  colours: string[];
+};
+
+const StylingView = ({ colours, environment, product, localSurvey, setLocalSurvey }: StylingViewProps) => {
+  const overwriteUnifiedStyling = useMemo(() => {
+    // unified styling is disabled from the product.
+    // we don't need to show the switch
+    if (!product.styling?.unifiedStyling) return true;
+
+    return !!localSurvey?.styling?.overwriteUnifiedStyling;
+  }, [localSurvey?.styling?.overwriteUnifiedStyling, product.styling?.unifiedStyling]);
+
+  const [formStylingOpen, setFormStylingOpen] = useState(false);
+  const [cardStylingOpen, setCardStylingOpen] = useState(false);
+  const [stylingOpen, setStylingOpen] = useState(false);
+
+  const setOverwriteUnifiedStyling = (value: boolean) => {
+    setLocalSurvey((prev) => ({ ...prev, styling: { ...prev.styling, overwriteUnifiedStyling: value } }));
+  };
+
+  const onResetUnifiedStyling = () => {
+    const { styling: productStyling } = product;
+    const { unifiedStyling, allowStyleOverwrite, ...baseStyling } = productStyling ?? {};
+
+    setLocalSurvey((prev) => ({
+      ...prev,
+      styling: {
+        ...baseStyling,
+        overwriteUnifiedStyling,
+      },
+    }));
+
+    toast.success("Styling set to unified styles");
+  };
+
+  useEffect(() => {
+    if (!overwriteUnifiedStyling) {
+      setFormStylingOpen(false);
+      setCardStylingOpen(false);
+      setStylingOpen(false);
+    }
+  }, [overwriteUnifiedStyling]);
+
+  useEffect(() => {
+    if (!product.styling?.unifiedStyling) {
+      setFormStylingOpen(true);
+    }
+  }, [product.styling?.unifiedStyling]);
+
+  return (
+    <div className="mt-12 space-y-3 p-5">
+      {!!product?.styling?.unifiedStyling && (
```

Comment:
I think we always have a product. Also I hope we can set `styling` to be mandatory in the product so it can't be null/undefined. This would make the check look nicer and more typesafe.